[DOCS] Clarify compatibility for upgrade via reindex (#60045)
elastic/elasticsearch: docs/reference/upgrade/reindex_upgrade.asciidoc

before upgrading to {version}. {es} nodes will fail to start if
incompatible indices are present. Snapshots of 5.x or earlier indices cannot be
restored to a 7.x cluster even if they were created by a 6.x cluster.

> Any index created in a 6.x version is 7.x compatible and do not require a reindex.

Reviewer: ```suggestion
Any index created in 6.x is compatible with 7.x and does not require a reindex.
```
